File: src/main/java/me/muksc/tacztweaks/mixininterface/feature/keyactions/unload/UnloadableGun.java
package me.muksc.tacztweaks.mixininterface.feature.keyactions.unload;

import com.tacz.guns.api.item.IGun;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.ItemStack;

public interface UnloadableGun {
    static UnloadableGun of(IGun instance) {
        return (UnloadableGun) instance;
    }

    void tacztweaks$unload(Player player, ItemStack gunItem);
}

File: src/main/kotlin/me/muksc/tacztweaks/mixininterop/UnloadableGun.kt
@file:Suppress("NOTHING_TO_INLINE")

package me.muksc.tacztweaks.mixininterop

import me.muksc.tacztweaks.mixininterface.feature.keyactions.unload.UnloadableGun
import net.minecraft.world.entity.player.Player
import net.minecraft.world.item.ItemStack

inline fun UnloadableGun.unload(player: Player, gunItem: ItemStack) =
    `tacztweaks$unload`(player, gunItem)